packages/venia-concept/src/util/getUrlKey.js

```diff
@@ -0,0 +1,17 @@
+import { basename } from "path";
+
+/**
+ * Extract the basename from a URL object, which corresponds to the `url_key`
+ * property of a Magento 2 GraphQL entity. Mostly used to get the url_key
+ * from a recently navigated-to URL.
+ *
+ * @param {URL} url
+ * @returns {string} A string for use as the `url_key` in a GraphQL query.
+ */
+export default function getUrlKey(url = window.location) {
```

Comment:
Don't think we need any of this - we can already get the product's ID for the product page query. The `urlResolver` query has a `canonical_url` field - this will always be in the format `catalog/product/view/id/{integer}`.

Reply:
Been meaning to do this anyway. Opened a PR to feed this to the RootComponent https://github.com/magento-research/pwa-studio/pull/93

Reply:
Thanks for that! I was halfway through writing that just now.

However, we still need the `url_key`, because of https://github.com/magento/graphql-ce/issues/86. We could also open an issue asking for the `urlResolver` to be an interface and we could add inline fragments for things like `sku`, but that would take longer than just doing it this way.